cmake/emil_build_for.cmake

```diff
@@ -0,0 +1,87 @@
+# 'emil_build_for' uses the EXCLUDE_FROM_ALL property on targets to specify whether the
+# target should be built for a certain combination of properties.
+#
+# These properties are supported:
```

Comment:
Should we describe that all properties are optional?

I.e. that you could also use the function like this:
`emil_buid_for(target HOST Linux)`

Reply:
I'll add that as an  example